File: js-pr/56/javascript/code-quality-readability-maintainability/data_transformer_17.js
// {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
function transformData(data) {
    let transformed = [];
    for (let item of data) {
        if (typeof item === 'object' && item !== null) {
            let newItem = {};
            for (let [key, value] of Object.entries(item)) {
                if (typeof value === 'string') {
                    newItem[key.toLowerCase()] = value.trim().replace(/\b\w/g, l => l.toUpperCase());
                } else if (typeof value === 'number') {
                    newItem[key.toLowerCase()] = value * 1.1;
                } else {
                    newItem[key.toLowerCase()] = value;
                }
            }
            transformed.push(newItem);
        }
    }
    return transformed;
}
// {/fact}

File: python-pr/52/python/code-quality-readability-maintainability/data_transformer_17.py
# {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
def transform_data(data):
    transformed = []
    for item in data:
        if isinstance(item, dict):
            new_item = {}
            for key, value in item.items():
                if isinstance(value, str):
                    new_item[key.lower()] = value.strip().title()
                elif isinstance(value, (int, float)):
                    new_item[key.lower()] = value * 1.1
                else:
                    new_item[key.lower()] = value
            transformed.append(new_item)
    return transformed
# {/fact}